src/Cimpress.Extensions.Http.Caching.InMemory/IMemoryCacheExtensions.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Threading.Tasks;
+using Cimpress.Extensions.Http.Caching.Abstractions;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace Cimpress.Extensions.Http.Caching.InMemory
+{
+    /// <summary>
+    /// Extension methods for an <see cref="IMemoryCache"/>.
+    /// </summary>
+    internal static class IMemoryCacheExtensions
+    {
+        /// <summary>
+        /// Tries to get the data from cache, that is, ignoring all exceptions.
+        /// </summary>
+        /// <param name="cache">The in memory cache.</param>
+        /// <param name="key">The key to retrieve from the cache.</param>
+        /// <returns>The data of the cache entry, or null if not found or on any error.</returns>
+        public static async Task<CacheData> TryGetAsync(this IMemoryCache cache, string key)
+        {
+            try
+            {
+                byte[] binaryData = null;
+                if (cache.TryGetValue(key, out binaryData))
+                {
+                    await Task.FromResult(true);
```

Comment:
I did this on purpose so that the implementation of the RedisCacheHandler and the InMemoryCacheHandler have the same code.  Theoretically one could extract the `SendAsync` methods now, and inject in these extensions as wrappers for the actual handlers.

But using the code like this, I've got parity between both handlers.  Should I take the last step to combine the classes as well.  I think it would go a long way to ensure they had the same interface.